src/nwsapi.js

```diff
@@ -1123,10 +1123,31 @@
                     ')){' + source + '}';
                   break;
                 case 'disabled':
-                  // https://www.w3.org/TR/html5/forms.html#enabling-and-disabling-form-controls:-the-disabled-attribute
-                  source = 'if((("form" in e||/^optgroup$/i.test(e.localName))&&"disabled" in e&&' +
-                    '(e.disabled===true||(n=s.ancestor("fieldset",e))&&(n=s.first("legend",n))&&!n.contains(e))' +
-                    ')){' + source + '}';
+                  // https://html.spec.whatwg.org/#enabling-and-disabling-form-controls:-the-disabled-attribute
+                  source = 'if((("form" in e||/^optgroup$/i.test(e.localName))&&"disabled" in e)){' +
+                    // F is true if any of the fieldset elements in the ancestry chain has the disabled attribute specified
+                    // L is true if the first legend element of the fieldset contains the element
+                    'var x=0,N=[],F=false,L=false;' +
+                    'if(!(/^(optgroup|option)$/i.test(e.localName))){' +
```

Comment:
This feels a bit messy, but `optgroup` has different definitions for `disabled`, as the HTML spec notes in the `disabled` attribute documentation:

> The [disabled](https://html.spec.whatwg.org/#attr-option-disabled) attribute for [option](https://html.spec.whatwg.org/#the-option-element) elements and the [disabled](https://html.spec.whatwg.org/#attr-optgroup-disabled) attribute for [optgroup](https://html.spec.whatwg.org/#the-optgroup-element) elements are defined separately.

We shouldn't take into account fieldset/legend for optgroup/option, so we skip over code related to that here.